### ISSUE TITLE
Need Dataset-Specific Shape Handling for Sampling Scripts - target_dim label generation fails

### ISSUE THREAD

**Issue by aduranu** [anirbansarkar-cs/D3-DNA-Discrete-Diffusion#26]
# Issue: Need Dataset-Specific Shape Handling for Sampling Scripts

## Problem Description

The current sampling scripts across different datasets (Promoter, DeepSTARR, MPRA, LenTimpra) use a uniform approach to handle user-specified conditioning labels that doesn't account for dataset-specific shape requirements or pre-shaped arrays.

**also, the use of signal_dim vs target_dim should be merged and clarified

### Current Implementation

All sampling scripts currently use the `.expand()` pattern for scalar/vector inputs:

```python
# Promoter (model_zoo/promoter/sample.py:113)
conditioning_labels = torch.tensor([[args.expression_target]], device=sampler.device).expand(args.num_samples, -1)

# DeepSTARR (model_zoo/deepstarr/sample.py:146)
conditioning_labels = torch.tensor([[args.dev_activity, args.hk_activity]], device=sampler.device).expand(args.num_samples, -1)

# MPRA (model_zoo/mpra/sample.py:96)
conditioning_labels = torch.tensor([[args.activity1, args.activity2, args.activity3]], device=sampler.device).expand(args.num_samples, -1)

# LenTimpra (model_zoo/lentimpra/sample.py:92)
conditioning_labels = torch.tensor([[args.activity]], device=sampler.device).expand(args.num_samples, -1)
```

### Issues

1. **No support for pre-shaped arrays**: If a user wants to pass a numpy array that already has shape `(num_samples, feature_dim)`, the current code will fail or produce incorrect results because:
   - `torch.tensor([[numpy_array]])` creates an unexpected nested structure
   - `.expand()` assumes a scalar/vector input and won't work correctly with multi-dimensional arrays

2. **Dataset-specific shape requirements**: Different datasets have different conditioning label shapes:
   - **Promoter**: Could be `(num_samples, 1)` for global expression or `(num_samples, seq_length, 1)` for per-position targets (transcription initiation signal)
   - **DeepSTARR**: `(num_samples, 2)` for (dev, hk) activities
   - **MPRA**: `(num_samples, 3)` for three regulatory activities
   - **LenTimpra**: `(num_samples, 1)` or potentially multi-class
   - **ATACseq**: Potentially different cell-type specific targets (up to 18)

3. fails to check if a generated label shape and target_dim are compatible with the checkpoint model

3. **Inconsistent handling**: The `generate_conditioning_labels()` method in each dataset's sampler already knows the correct shape, but the user-specified path doesn't leverage this knowledge.

### Example Failure Case

If a user tries to pass a numpy array of shape `(num_samples, 1024)` for Promoter (e.g., per-position expression targets):

```python
# This would fail with current code:
args.expression_target = np.array([[0.5] * 1024] * num_samples)  # shape: (num_samples, 1024)
conditioning_labels = torch.tensor([[args.expression_target]], device=sampler.device).expand(args.num_samples, -1)
# Results in unexpected shape, likely (1, 1, num_samples, 1024) or error
```

## Proposed Solution

1. **Add shape detection and handling** in each dataset's sampling script:
   - Check if the input is already a numpy array/tensor with the correct shape
   - If it's a scalar/vector, use the current `.expand()` pattern
   - If it's already shaped correctly, convert directly to tensor

2. **Leverage dataset-specific knowledge**:
   - Use the `generate_conditioning_labels()` method to determine expected shape
   - Add validation to ensure user-provided labels match expected dimensions
   - Support both global and per-position conditioning where applicable

3. **Example implementation pattern**:

```python
if args.expression_target is not None:
    if isinstance(args.expression_target, (np.ndarray, torch.Tensor)):
        # Already has shape - validate and convert
        if isinstance(args.expression_target, np.ndarray):
            conditioning_labels = torch.tensor(args.expression_target, device=sampler.device)
        else:
            conditioning_labels = args.expression_target.to(sampler.device)
        
        # Validate shape
        expected_shape = sampler.generate_conditioning_labels(1, config).shape[1:]
        if conditioning_labels.shape[1:] != expected_shape:
            raise ValueError(f"Expected conditioning label shape (num_samples, {expected_shape}), "
                           f"got {conditioning_labels.shape}")
        if conditioning_labels.shape[0] != args.num_samples:
            raise ValueError(f"Expected {args.num_samples} samples, got {conditioning_labels.shape[0]}")
    else:
        # Scalar/vector value - expand it
        conditioning_labels = torch.tensor([[args.expression_target]], device=sampler.device).expand(args.num_samples, -1)
```

## Affected Files

- `model_zoo/promoter/sample.py` (line 113)
- `model_zoo/deepstarr/sample.py` (line 146)
- `model_zoo/mpra/sample.py` (line 96)
- `model_zoo/lentimpra/sample.py` (line 92)
- Potentially other dataset sampling scripts

## Benefits

1. **Flexibility**: Users can pass either scalar values or pre-shaped arrays
2. **Dataset-specific**: Each dataset can handle its own shape requirements
3. **Validation**: Early error detection for shape mismatches
4. **Consistency**: Better alignment with how `generate_conditioning_labels()` works

## Priority

Medium - This is a usability improvement that would make the sampling scripts more flexible and robust, especially for advanced use cases where users want to condition on specific label arrays.
